core/extensions/impl/storage_extension.cpp

```diff
@@ -280,21 +281,23 @@ namespace kagome::extensions {
       runtime::WasmSpan key) {
     auto [key_ptr, key_size] = runtime::WasmResult(key);
     auto key_buffer = memory_->loadN(key_ptr, key_size);
-    auto data = get(key_buffer);
-    if (not data) {
-      logger_->trace("ext_get_storage_into. Val by key {} not found",
-                     key_buffer.toHex());
-      return runtime::WasmMemory::kMaxMemorySize;
-    }
-    if (not data.value().empty()) {
-      logger_->trace("ext_get_storage_into. Key hex: {} , Value hex {}",
+
+    auto result = get(key_buffer);
+    auto option = result ? boost::make_optional(result.value()) : boost::none;
+
+    if (option) {
+      logger_->trace("ext_storage_get_version_1( {} ) => {}",
                      key_buffer.toHex(),
-                     data.value().toHex());
+                     option.value().empty() ? "empty" : option.value().toHex());
+
     } else {
-      logger_->trace("ext_get_storage_into. Key hex: {} Value: empty",
+      logger_->trace("ext_storage_get_version_1( {} ) => not found",
                      key_buffer.toHex());
     }
-    return memory_->storeBuffer(data.value());
+
+    auto encoded = scale::encode(option).value();
+
+    return memory_->storeBuffer(encoded);
```

Comment:
rval?
return memory_->storeBuffer(scale::encode(option).value());

Reply:
Also this code will be executed in the case ```!option```. Is it wright?

Reply:
I mean ```.value()``` will be failed.

Reply:
> Also this code will be executed in the case !option. Is it wright?

yes

Reply:
> I mean ```.value()``` will be failed.

scale::encode never fails)) we should change signature of this function later